### DISCUSSION TITLE
allow to expose docker socket via port

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#3230] [Ideas] [answered]
- add ecosystem to startup apps
- add checkbox to aio inteface, disable by default
- if enabled, protect with aio_secret, and/or only allow access from the nextcloud container. Question is how we can proxy that? Probably via apache? but then how to handle the authentication?

See https://github.com/cloud-py-api/app_ecosystem_v2/blob/main/Makefile for reference

**Comment by Zoey2936** [comment]
can I ask why this is needed?

**Reply by szaimen** [reply]
So the idea is that the app-ecosystem v2 allows to start docker containers itself. Since AIO is built around docker it would make the install much easier to allow to expose the socket via a port so that the system can manage the things itself. 

I was unfortunately included into the project too late to actually make sure that the logic that is built into AIO makes it into the app-ecosystem itself. So the easier way for us now is to simply add this feature (which is disabled by default because of security implications) and make sure that one can enable it if needed. 

Also see https://github.com/cloud-py-api/app_ecosystem_v2/tree/main for reference.

**Reply by Zoey2936** [reply]
so you want something like [this](https://github.com/ZoeyVid/docker-socket-proxy/) (the README is outdated), inbuilt to the mastercontainer?

**Reply by szaimen** [reply]
Indeed. Thanks for the link! I will look into that going forward :)

**Comment by szaimen** [comment]
Another idea would be to actually put this into a separate container. Then we could for example even reuse the docker-socket-proxy container. In that case We should probably actually put the mastercontainer, the docker-socket-proxy and the nextcloud container into a dedicated network.

**Comment by szaimen** [comment, marked as answer]
Will be done in https://github.com/nextcloud/all-in-one/pull/3235

**Comment by gggeek** [comment]
Related discussion: https://github.com/nextcloud/all-in-one/discussions/1713#discussioncomment-16974508
